spectral_cube/lower_dimensional_structures.py

```diff
@@ -1093,6 +1120,51 @@ def __getattribute__(self, attrname):
 
 class VaryingResolutionOneDSpectrum(OneDSpectrum, MultiBeamMixinClass):
     def __new__(cls, value, beams=None, **kwargs):
+        out = super(VaryingResolutionOneDSpectrum, cls).__new__(cls, value, **kwargs)
         if beams is not None:
-            cls.beams = beams
-        return super(VaryingResolutionOneDSpectrum, cls).__new__(cls, value, **kwargs)
+            out.beams = beams
+        return out
+
+    def with_beams(self, beams):
+        '''
+        Attach a new beam object to the OneDSpectrum.
+
+        Parameters
+        ----------
+        beams : `~radio_beam.Beams`
+            A new beams object.
+        '''
+
+        meta = self.meta.copy()
+        meta['beams'] = beams
+
+        return self._new_spectrum_with(beams=beams, meta=meta)
+
+    def with_beam(self, beam):
```

Comment:
I wanted to block the case of setting a single beam to a VRODS so there was no way to mistakenly go from VRODS to ODS. Raising the error could be an issue though... same with the re-defined `beam` parameter below.